### PR TITLE
Adding block update event handler

### PR DESCRIPTION
**Pull request by fishseabowl**
## Description
Fixes pyrsia/pyrsia#832

This PR is implementing the block update event. 

## PR Checklist

- [X] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [X] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [X] I've included a good title and brief description along with how to review them.
- [X] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [X] I've built the code `cargo build --all-targets` successfully.
- [X] I've run the unit tests `cargo test --workspace` and everything passes.
- [X] I've made sure my rust toolchain is current `rustup update`.

### CODE REVIEW

File: pyrsia_node/src/network/handlers.rs [hunk 1]
```diff
@@ -66,3 +68,18 @@ pub async fn handle_request_idle_metric(
...
+    blockchain: Arc<Mutex<Blockchain>>,
+    block_ordinal: u64,
+    block: Vec<u8>,
+) -> anyhow::Result<()> {
+    debug!(
+        "Handling request blockchain: {:?}={:?}",
+        block_ordinal, block
+    );
+
+    let _ = blockchain; // TODO!
```
Comment (johanvos) [inline pyrsia_node/src/network/handlers.rs:83]: Is this TODO already defined in an issue? Otherwise, it would be good if you can create an issue for it.

File: src/network/blockchain_protocol.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
+/*
+   Copyright 2021 JFrog Ltd
```
Comment (johanvos) [inline src/network/blockchain_protocol.rs:2]: 2022

File: src/network/blockchain_protocol.rs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+#[derive(Clone)]
+pub struct BlockUpdateExchangeCodec();
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct BlockUpdateRequest(pub u64, pub Vec<u8>);
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct BlockUpdateResponse();
+
+impl ProtocolName for BlockUpdateExchangeProtocol {
+    fn protocol_name(&self) -> &[u8] {
+        "/pyrsia-blockchain-update-exchange/1".as_bytes()
```
Comment (johanvos) [inline src/network/blockchain_protocol.rs:38]: might be simpler to use v1,... for version numbers. But afaik, this is not used anywhere else yet?

File: pyrsia_node/src/network/handlers.rs (hunk 1 above)
Comment (fishseabowl) [reply pyrsia_node/src/network/handlers.rs:83]: Thanks. I will add some issues for this. Others, this is part of issue #780.

File: src/network/blockchain_protocol.rs (hunk 3 above)
Comment (fishseabowl) [reply src/network/blockchain_protocol.rs:38]: Thanks. Valid names are better. Ref: https://docs.rs/libp2p/latest/libp2p/core/trait.ProtocolName.html

**Review by tiainen — APPROVED** [review]

File: pyrsia_node/src/network/handlers.rs (hunk 1 above)
Comment (efrisch) [inline pyrsia_node/src/network/handlers.rs:83]: Why is this here?

**Review by efrisch — APPROVED** [review]
👍
